contrib/cindercompatibleapi/converter/volume.go

```diff
@@ -0,0 +1,380 @@
+type CreateVolumeReqSpec struct {
+	Volume         CreateReqVolume `json:"volume"`
+	SchedulerHints SchedulerHints  `json:"OS-SCH-HNT:scheduler_hints,omitempty"`
+}
+
+// CreateReqVolume ...
+type CreateReqVolume struct {
+	Size               int64             `json:"size"`
+	AvailabilityZone   string            `json:"availability_zone,omitempty"`
+	SourceVolID        string            `json:"source_volid,omitempty"`
+	Description        string            `json:"description,omitempty"`
+	Multiattach        bool              `json:"multiattach,omitempty"`
+	SnapshotID         string            `json:"snapshot_id,omitempty"`
+	BackupID           string            `json:"backup_id,omitempty"`
+	Name               string            `json:"name"`
+	ImageRef           string            `json:"imageRef,omitempty"`
+	VolumeType         string            `json:"volume_type,omitempty"`
+	Metadata           map[string]string `json:"metadata,omitempty"`
+	ConsistencygroupID string            `json:"consistencygroup_id,omitempty"`
+}
+
+// SchedulerHints ...
+type SchedulerHints struct {
+	SameHost []string `json:"same_host,omitempty"`
+}
+
+// CreateVolumeRespSpec ...
+type CreateVolumeRespSpec struct {
+	Volume CreateRespVolume `json:"volume,omitempty"`
+}
+
+// CreateRespVolume ...
+type CreateRespVolume struct {
+	MigrationStatus    string            `json:"migration_status,omitempty"`
+	Attachments        []RespAttachment  `json:"attachments"`
+	Links              []Link            `json:"links,omitempty"`
+	AvailabilityZone   string            `json:"availability_zone,omitempty"`
+	Encrypted          bool              `json:"encrypted ,omitempty"`
+	UpdatedAt          string            `json:"updated_at,omitempty"`
+	ReplicationStatus  string            `json:"replication_status,omitempty"`
+	SnapshotID         string            `json:"snapshot_id,omitempty"`
+	ID                 string            `json:"id,omitempty"`
+	Size               int64             `json:"size,omitempty"`
+	UserID             string            `json:"user_id,omitempty"`
+	Metadata           map[string]string `json:"metadata"`
+	Status             string            `json:"status,omitempty"`
+	Description        string            `json:"description,omitempty"`
+	Multiattach        bool              `json:"multiattach,omitempty"`
+	SourceVolID        string            `json:"source_volid,omitempty"`
+	ConsistencygroupID string            `json:"consistencygroup_id,omitempty"`
+	Name               string            `json:"name,omitempty"`
+	Bootable           bool              `json:"bootable,omitempty"`
+	CreatedAt          string            `json:"created_at,omitempty"`
+	VolumeType         string            `json:"volume_type,omitempty"`
+}
+
+// Link ...
+type Link struct {
+	Href string `json:"href,omitempty"`
+	Rel  string `json:"rel,omitempty"`
+}
+
+// CreateVolumeReq ...
+func CreateVolumeReq(cinderReq *CreateVolumeReqSpec) (*model.VolumeSpec, error) {
```

Comment:
I don't think this method should return error, since other methods only return ***RespSpec. You can package the validate operation in a standalone method.